addons/mail/static/src/services/new_message_service/new_message_service.js

```diff
@@ -0,0 +1,20 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+
+const { Component } = owl;
+
+export const newMessageService = {
+    dependencies: ["command"],
+    start(newEnv, { command }) {
+        command.add("Message a user", () => {
+            const env = Component.env;
+            env.messaging.chatWindowManager.openNewMessage();
```

Comment:
still need to add the 2 await as on the other PR, without them using shortcut before messaging is created would lead to a crash

Reply:
I don't understand the use case ? openNewMessage isn't async ?

Reply:
I meant
```
        await this.env.messagingCreatedPromise;
        await this.env.messaging.initializedPromise;
```

Reply:
Calling `openNewMessage()` only works if discuss app is not open. If discuss app is open, we'll need to call a new equivalent method to the currently existing `onClickMobileNewChatButton`.

Reply:
How to know if discuss app is open ?

Reply:
If I want to do something like `onClickMobileNewChatButton`, I need to have access to discuss.How can I do this?

Reply:
`env.messaging.discuss.isOpen`